src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -203,12 +204,55 @@
         type="function_name_identifier",
     ),
     QuotedLiteralSegment=OneOf(
-        NamedParser("single_quote", CodeSegment, name="quoted_literal", type="literal"),
-        NamedParser(
-            "unicode_single_quote", CodeSegment, name="quoted_literal", type="literal"
+        # Postgres allows newline-concatenated string literals (#1488).
+        # Since these string literals can have comments between them,
+        # we use grammar to handle this.
+        Delimited(
+            NamedParser(
+                "single_quote",
+                CodeSegment,
+                name="quoted_literal",
+                type="literal",
+            ),
+            delimiter=NamedParser(
+                "newline",
+                NewlineSegment,
+                name="newline",
+                type="newline",
+            ),
+            allow_trailing=True,
```

Comment:
Postgres is a little more complicated: quoted literals must have at least one newline between them (additional whitespace is fine) to be concatenated and therefore parse. This is done by delimiting via newline segments. The `allow_trailing` is because in this context we don't want to fail parsing if there's a newline after the final concatenated quoted literal, that's quite standard.

(N.B. `unicode_single_quote` and `escaped_single_quote` follow the same logic)

Reply:
Also note redshift inherits this same logic and @rpr-ableton has kindly confirmed this behaviour for us [here](https://github.com/sqlfluff/sqlfluff/issues/1488#issuecomment-1016664179)